concepts/methods/about.md

```diff
@@ -0,0 +1,152 @@
+# About
+
+In Rust, methods are functions defined for a struct (or an `enum` or `trait` object).
+This changes the calling pattern: instead of `frobnicate(foo)`, a method is called as `foo.frobnicate()`.
+
+Methods are similar to functions: they are declared with the `fn` keyword,
+they have a name, and they can have parameters and return values.
+
+Unlike functions, a methods's first parameter is always `self` (or a variant),
+which represents the instance of the struct the method is being called on (more details below).
+
+Let's create a `enum` for Hogwarts's houses and a `struct` that describes a wizard's `age` and `house`:
+
+```rust
+#[derive(Debug)]
+enum House {
+    Gryffindor,
+    Hufflepuff,
+    Ravenclaw,
+    Slytherin,
+}
+
+struct Wizard {
+    age: u32,
+    house: House,
+}
+```
+
+## Defining methods
+
+To define methods for the struct `Wizard`, we start an `impl` (short for implementation) block for `Wizard`.
+
+```rust
+impl Wizard { }
+```
+
+Everything within this `impl` block will be associated with the `Wizard` type.
+Note that a struct is allowed to have multiple `impl` blocks.
+
+## Using `self`
+
+A method must have a parameter named `self` of type `Self` as its first parameter.
+The type `Self` is an alias for the type that the `impl` block is for.
+In our example, `Self` represents `Wizard`.
+
+Rust allows us to abbreviate this with only the name `self` in the first parameter spot for convenience.
+Common `self` types for methods include:
+
+- `self`: Abbreviates `self: Self`. The bare `self` signature consumes the instance
+- `&self`: Abbreviates `self: &Self`. This borrows the instance immutably.
+- `&mut self`: Abbreviates `self: &mut Self`. This borrows the instance mutably.
+
+In more advanced code you might also see signatures such as `self: Pin<&self>`, `self: Rc<Self>`, etc.
+This is, however, beyond the scope of the current tutorial.
+
+### Defining method with `&self`
+
+If we wish to implement an `info` method to display the basic information of the wizard,
+we could define this method inside an `impl` block for `Wizard`:
+
+```rust
+impl Wizard {    
+    fn info(&self) {
+        println!(
+            "A wizard of age {} who studies in House {:?} at Hogwarts",
+            self.age,
+            self.house,
+        );
+    }
+}
+```
+
+Note that in the signature for `info`, we borrow the instance immutably with `&self`.
+This way, we could read the `age` and `house` fields of a `Wizard` instance without changing them.
```

Comment:
```suggestion
This way, we could read the `age` and `house` fields of a `Wizard` instance without changing them.
If we had used a consuming `self` instead of borrowing, the instance would be inaccessable after the method was called.
```

It's a bit redundant, but I think we should really emphasize that most of the time you want `&self`.